### PR TITLE
Optimize nginx & recordings

### PR DESCRIPTION
**Pull request by NickM-27**
• Use faststart for recordings segments
• Use threading in nginx
• Reload video on first error while viewing recordings

### CODE REVIEW

**Comment by blakeblackshear** [comment]
A couple tings we should look into:

- providing `keyFrameDurations` for the clips from the database (would need to add a field to the db) 
- using the concat clip type

**Comment by NickM-27** [comment]
> I don't think `keyFrameDurations` looks useful. Maybe setting discontinuity to false could be helpful. 

Isn't it already false? https://github.com/blakeblackshear/frigate/blob/dev/frigate/http.py#L1139 or do you mean setting that somewhere else too?

**Comment by NickM-27** [comment]
Tried out concat but it doesn't seem to make a difference, also found this: https://github.com/kaltura/nginx-vod-module/issues/264#issuecomment-203808545

> Concat filter is very similar to using multiple clips with discontinuity set to false, the added value it provides is that the duration of the mp4s can be different across different sequences.

So seems it may not be relevant since we only have one sequence

**Comment by NickM-27** [comment]
Dash seems to be working better for me. 

HLS:

Screen Shot 2022-12-13 at 14 30 02 PM

DASH: 

Screen Shot 2022-12-13 at 14 39 15 PM

**Comment by NickM-27** [comment]
Can't move forward with DASH as it is not supported on iOS

File: Dockerfile [hunk 1]
```diff
@@ -79,8 +126,8 @@ RUN wget -qO edgetpu_model.tflite https://github.com/google-coral/test_data/raw/
 RUN wget -qO cpu_model.tflite https://github.com/google-coral/test_data/raw/release-frogfish/ssdlite_mobiledet_coco_qat_postprocess.tflite
 COPY labelmap.txt .
 # Copy OpenVino model
-COPY --from=ov-converter /models/public/ssdlite_mobilenet_v2/FP16 openvino-model
-RUN wget -q https://github.com/openvinotoolkit/open_model_zoo/raw/master/data/dataset_classes/coco_91cl_bkgr.txt -O openvino-model/coco_91cl_bkgr.txt
+#COPY --from=ov-converter /models/public/ssdlite_mobilenet_v2/FP16 openvino-model
```
Comment (blakeblackshear) [inline Dockerfile:129]: Did you mean to comment this out?

File: docker-compose.yml [hunk 2]
```diff
@@ -12,16 +12,16 @@ services:
...
+#      - /dev/bus/usb:/dev/bus/usb
       # - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
     volumes:
       - .:/workspace/frigate:cached
       - ./web/dist:/opt/frigate/web:cached
       - /etc/localtime:/etc/localtime:ro
       - ./config/config.yml:/config/config.yml:ro
       - ./debug:/media/frigate
-      - /dev/bus/usb:/dev/bus/usb
+#      - /dev/bus/usb:/dev/bus/usb
```
Comment (blakeblackshear) [inline docker-compose.yml:24]: Do we want to leave these out by default now?

File: docker/rootfs/usr/local/nginx/conf/nginx.conf [hunk 3]
```diff
@@ -61,6 +64,19 @@ http {
         vod_align_segments_to_key_frames on;
         vod_manifest_segment_durations_mode accurate;
         vod_ignore_edit_list on;
+        vod_segment_duration 10000;
+        vod_hls_mpegts_align_frames off;
+        vod_hls_mpegts_interleave_frames on;
+
+		# file handle caching / aio
+		open_file_cache          max=1000 inactive=5m;
```
Comment (blakeblackshear) [inline docker/rootfs/usr/local/nginx/conf/nginx.conf:72]: The indentation in the config is inconsistent.

**Review by blakeblackshear — CHANGES_REQUESTED** [review]
One other thing we could do is prefetch the next hour's playlist ahead of time so VOD will populate the metadata cache. That may get rid of the loading when going to the next hour during playback.

File: Dockerfile (hunk 1 above)
Comment (NickM-27) [reply Dockerfile:129]: For some reason my M1 mac is missing an AVX arg to do this conversion, not really sure why. Anyway no I didn't mean for that to get committed

File: docker-compose.yml (hunk 2 above)
Comment (NickM-27) [reply docker-compose.yml:24]: I'm not sure it matters for people developing on linux, it just complains on my mac. Might be worth considering but probably separate from this PR

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**